token-metadata/program/src/instruction.rs

```diff
@@ -544,38 +544,42 @@ pub enum MetadataInstruction {
+    #[account(7, name="sysvar_instructions", desc="Instructions sysvar account")]
+    #[account(8, optional, signer, name="authority", desc="Authority/creator of the escrow account")]
     CreateEscrowAccount,
 
     /// Close the escrow account.
     #[account(0, writable, name="escrow", desc="Escrow account")]
-    #[account(1, name="metadata", desc="Metadata account")]
+    #[account(1, writable, name="metadata", desc="Metadata account")]
     #[account(2, name="mint", desc="Mint account")]
     #[account(3, name="token_account", desc="Token account")]
     #[account(4, name="edition", desc="Edition account")]
     #[account(5, writable, signer, name="payer", desc="Wallet paying for the transaction and new account")]
     #[account(6, name="system_program", desc="System program")]
+    #[account(7, name="sysvar_instructions", desc="Instructions sysvar account")]
     CloseEscrowAccount,
 
     /// Transfer the token out of Escrow.
     #[account(0, name="escrow", desc="Escrow account")]
-    #[account(1, writable, signer, name="payer", desc="Wallet paying for the transaction and new account")]
-    #[account(2, name="attribute_mint", desc="Mint account for the new attribute")]
-    #[account(3, writable, name="attribute_src", desc="Token account source for the new attribute")]
-    #[account(4, writable, name="attribute_dst", desc="Token account, owned by TM, destination for the new attribute")]
-    #[account(5, name="escrow_mint", desc="Mint account that the escrow is attached")]
-    #[account(6, name="escrow_account", desc="Token account that holds the token the escrow is attached to")]
-    #[account(7, name="system_program", desc="System program")]
-    #[account(8, name="ata_program", desc="Associated Token program")]
-    #[account(9, name="token_program", desc="Token program")]
-    #[account(10, name="rent", desc="Rent info")]
-    #[account(11, optional, signer, name="authority", desc="Authority/creator of the escrow account")]
+    #[account(1, writable, name="metadata", desc="Metadata account")]
+    #[account(2, writable, signer, name="payer", desc="Wallet paying for the transaction and new account")]
+    #[account(3, name="attribute_mint", desc="Mint account for the new attribute")]
+    #[account(4, writable, name="attribute_src", desc="Token account source for the new attribute")]
+    #[account(5, writable, name="attribute_dst", desc="Token account, owned by TM, destination for the new attribute")]
+    #[account(6, name="escrow_mint", desc="Mint account that the escrow is attached")]
+    #[account(7, name="escrow_account", desc="Token account that holds the token the escrow is attached to")]
+    #[account(8, name="system_program", desc="System program")]
+    #[account(9, name="ata_program", desc="Associated Token program")]
+    #[account(10, name="token_program", desc="Token program")]
+    #[account(11, name="rent", desc="Rent info")]
```

Comment:
We do not need the rent account, we can delete it and use the `instruction::create_associated_account` which does not require rent passed in. The version of that instruction currently used is deprecated.

Reply:
Good call. I believe this was made before the TM dependency update.